### DISCUSSION TITLE
CSS Source maps not working in development [Next 13]

### DISCUSSION THREAD

**Discussion by rodsotdia** [vercel/next.js#47887] [Help]
### Summary

I'm using NextJS `13.2.4` and I've installed Tailwind [according to Next docs](https://beta.nextjs.org/docs/styling/tailwind-css) and configured sass [as per the docs](https://beta.nextjs.org/docs/styling/sass#customizing-sass-options), but I can't get the source maps working, to use the dev tools and see what are the sources of the CSS written in development. I have all my `.scss` files in a folder, and everything is working well, but at the moment to inspect CSS the source file is `http://localhost:3000/_next/static/css/_app-client_src_styles_main_scss.css` making it hard to debug.

My dev tools have the option enabled to see CSS source maps, and it looks like the source maps files don't exist. I need the source maps for development purposes using `npm run dev`. The source maps only work for production, after doing `npm run build` and then 'npm run start', but not for dev.

How can I enable the source maps for development?

This is my `next.config.js`:
```
/** @type {import('next').NextConfig} */
const path = require("path");

const nextConfig = {
  experimental: {
    appDir: true,
  },
  sassOptions: {
    includePaths: [path.join(__dirname, "styles")],
    sourceMap: true,
  },
  productionBrowserSourceMaps: true,
};

module.exports = nextConfig;

```

Any help is appreciated!

### Additional information

```js
Dependencies details:

"dependencies": {
    "eslint": "8.36.0",
    "eslint-config-next": "13.2.4",
    "gsap": "file:gsap-bonus.tgz",
    "next": "13.2.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-transition-group": "^4.4.5",
    "zustand": "^4.3.6"
  },
  "devDependencies": {
    "autoprefixer": "^10.4.14",
    "postcss": "^8.4.21",
    "prettier": "^2.8.6",
    "sass": "^1.59.3",
    "tailwindcss": "^3.2.7"
  },
```

**Comment by K3vinF** [comment]
I can confirm this.
In dev mode node no css sourcemaps are available.
If I run a build with ` productionBrowserSourceMaps: true` in next.config.js then the sourcemaps do work.

**Comment by najibghadri** [comment]
Same, can confirm.

**Comment by rodsotdia** [comment]
Anyone could fix this? Using NextJS `13.5.1` and the issue persists.

**Reply by leodutra** [reply]
The alternative for me was to use react extension for dev tools for my needs. I couldn't fix it after some tries.

**Reply by rodsotdia** [reply]
Thanks for your comment @leodutra 
Apparently, this doesn't work for me, as I need to see the source of the CSS (.scss files).

**Reply by leodutra** [reply]
![image](https://github.com/vercel/next.js/assets/444344/fa9265e5-418e-4c7a-84d9-ad25ae7ce70a)
Have you checked if it is not under webpack _Nsomething? If I remember well, it should expose scss too.

**Reply by leodutra** [reply]
look for _Nsomething and webpack:// in your dev tool sources leftmost tab

**Reply by rodsotdia** [reply]
@leodutra Expose a lot of things, including all the component files, but not the folder with the `.scss` files.

**Reply by MarkWang2** [reply]
Still has the issue.

**Comment by adleviton** [comment]
Also having this issue in Next 14

**Comment by 9vitorconti** [comment]
Same here on nextjs 14

**Comment by rodsotdia** [comment]
Just confirming that this is still not working on `14.0.2`

**Comment by marcelhageman** [comment]
Same still for `14.0.4`.

It's very frustrating to inspect elements and see the parsed CSS files only. I want to see the unparsed SCSS files.

A different project of mine uses `13.4.2` and it works fine there. Tried playing with postcss settings and such, no change.

**Comment by rodsotdia** [comment]
For everyone landing here, I've found a solution to inspect `.scss` files. It looks like they disabled the option intentionally, due to performance stuff, but with the following configuration in `next.config.js` you can enable it:

```
const nextConfig = {
  sassOptions: {
    includePaths: [path.join(__dirname, "styles")],
  },
  webpack: (config, options) => {
    /**
     * Force scss source maps for debugging. If there are performance issues or you don't need debug css, use the value "eval-source-map" instead.
     */
    if (options.dev) {
      Object.defineProperty(config, "devtool", {
        get() {
          return "source-map";
        },
        set() {},
      });
    }

    return config;
  },
};
```

You will get a warning on your terminal, but it does work. At the moment, it is the only way I've found to debug `.scss`, hopefully, this get resolved in future versions.

**Reply by developerr-ayush** [reply]
hey it says path not available

**Reply by rodsotdia** [reply]
@developerr-ayush Double check the `includePaths` path, in the above case, there is a styles folder where I have all `.scss` files, maybe your case is different.

**Reply by developerr-ayush** [reply]
I have done * instead of path.join(__dirname, "styles") and resolved

**Reply by MilanCiprian** [reply]
Add above `const nextConfig`

```
import path from 'path';
import { fileURLToPath } from 'url';
const __filename = fileURLToPath(import.meta.url); // get the resolved path to the file
const __dirname = path.dirname(__filename); // get the name of the directory

```

**Reply by shubham-markobrando** [reply]
thanks, It's working.
